kyaml/yaml/serialization.go

```diff
@@ -34,43 +34,12 @@ func DoSerializationHacks(node *yaml.Node) {
 			// https://github.com/go-yaml/yaml/issues/587 in go-yaml.v3
 			// Remove this hack when the issue has been resolved
 			if len(node.Content) > 0 && node.Content[0].Kind == ScalarNode {
-				node.HeadComment = node.Content[0].HeadComment
+				// Don't clobber the head comment if it's not empty.
+				if node.HeadComment == "" && node.Content[0].HeadComment != "" {
+					node.HeadComment = node.Content[0].HeadComment
+				}
 				node.Content[0].HeadComment = ""
 			}
 		}
 	}
 }
-
-func UndoSerializationHacksOnNodes(nodes []*RNode) {
```

Comment:
Why are you removing this ? This is to make sure the resources are not modified due to serialization hacks.

Reply:
Writer should not modify the input nodes.
I copy the input nodes and let the writer to whatever it want on the copy. So we don't need `UndoSerializationHacksOnNodes` anymore. See:
https://github.com/kubernetes-sigs/kustomize/pull/3965/files#diff-fd61368f626f1a6faa338f8665a53c99bb5fab47449926925767790a64839f30R53-R164

Also I found we invokes `DoSerializationHacks` in `UndoSerializationHacks`. I'm not sure if it's a bug or not.

Reply:
Copying works.